src/App.js

```diff
@@ -1,25 +1,86 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { useState } from 'react'; 
+import contacts from './contacts.json';
 import './App.css';
 
+const contactState = contacts.splice(0,5);
+
 function App() {
+  const [ contact, setContact ] = useState(contactState)
+
+  const randomContactHandler = () => {
+    const randomNumber = Math.floor(Math.random() * contacts.length);
+    const randomContact = contacts[randomNumber]
+    const contactsCopy = [...contact];
+        contactsCopy.push(randomContact);
+        setContact(contactsCopy); 
+  };
+
+  const sortByName = () => {
+    const sortedNames = [...contact];
+
+    sortedNames.sort((a, b) => {
+        let nameA = a.name.toUpperCase();
+        let nameB = b.name.toUpperCase()
+        if(nameA < nameB) {
+          return -1;
+        } if(nameA > nameB) {
+          return 1;
+        } else {return 0;}
+      });
+
+      setContact(sortedNames); 
+
+  }
+
+  const sortByPop = () => {
+    const sortedPop = [...contact];
+
+    sortedPop.sort((a, b) => {
+        let popA = a.popularity
+        let popB = b.popularity
+        if(popA > popB) {
+          return -1;
+        } if(popA < popB) {
+          return 1;
+        } else {return 0;}
+      });
+
+      setContact(sortedPop); 
+  }
+
+  const deleteMovie = (contactId) => {
+    const contactsCopy = [...contact];
+    const contactIndex = contactsCopy.findIndex((contact) => contact.id === contactId);
+    contactsCopy.splice(contactIndex, 1)
+    setContact(contactsCopy)
+  }
+
+
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <h1>My Awesome Contacts</h1>
+      <div>
+        <button onClick={() => randomContactHandler()} className="app-btns">Add Random Actor</button>
+        <button onClick={() => sortByName()} className="app-btns">Sort by Name</button>
+        <button onClick={() => sortByPop()} className="app-btns">Sort by Popularity</button>
+      </div>
+      <table className="table table-row">
+      <tr>
+        <th>Picture</th>
+        <th>Name</th>
+        <th>Popularity</th>
+        <th>Action</th>
+      </tr>
+      {contact.map((contactItem) => 
+        <tr>
```

Comment:
When you are rendering a list, React wants you to use a `key` attribute on the element with a unique identifier.
```
<tr key={contactItem.id}>
   ...
</tr>